config/routes.rb

```diff
@@ -1,8 +1,22 @@
 Rails.application.routes.draw do
 
-  resources :companies do
-    resources :jobs
+  root to: "dashboard#index"
+
+  resources :dashboard, only: [:index]
+
+  resources :jobs, only: [:index]
+
+  resources :companies, only: [:index, :new, :create, :edit, :update, :destroy] do
+    resources :contacts, only: [:create]
+    resources :jobs, :controller => "companies/jobs" do
+      resources :comments, only: [:create]
```

Comment:
whenever you find yourself nesting three deep, reconsider if this is the best way to do things